Makefile

```diff
@@ -0,0 +1,352 @@
+# ====================================================================================
+# Variables
+
+## General Variables
+# Branch Variables
+PROTECTED_BRANCH := main
+CURRENT_BRANCH   := $(shell git rev-parse --abbrev-ref HEAD)
+# Use repository name as application name
+APP_NAME    := $(shell basename -s .git `git config --get remote.origin.url`)
+# Get current commit
+APP_COMMIT  := $(shell git log --pretty=format:'%h' -n 1)
+# Check if we are in protected branch, if yes use `protected_branch_name-sha` as app version.
+# Else check if we are in a release tag, if yes use the tag as app version, else use `dev-sha` as app version.
+APP_VERSION ?= $(shell if [ $(PROTECTED_BRANCH) = $(CURRENT_BRANCH) ]; then echo $(PROTECTED_BRANCH)-$(APP_COMMIT); else (git describe --abbrev=0 --exact-match --tags 2>/dev/null || echo dev-$(APP_COMMIT)) ; fi)
+
+# Get current date and format like: 2022-04-27 11:32
+BUILD_DATE  := $(shell date +%Y-%m-%d\ %H:%M)
+
+## General Configuration Variables
+# We don't need make's built-in rules.
+MAKEFLAGS     += --no-builtin-rules
+# Be pedantic about undefined variables.
+MAKEFLAGS     += --warn-undefined-variables
+# Set help as default target
+.DEFAULT_GOAL := help
+
+# App Code location
+CONFIG_APP_CODE         += ./cmd/rtcd
```

Comment:
This will probably needs to be changed to point to the code on this repo. Just noting

Reply:
Changed to code folder.